boards/litex_arty/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+name = "litex_arty"
+version = "0.1.0"
+authors = ["Tock Project Developers <tock-dev@googlegroups.com>"]
+build = "build.rs"
+edition = "2018"
+
+[dependencies]
+components = { path = "../components" }
+rv32i = { path = "../../arch/rv32i" }
+capsules = { path = "../../capsules" }
+kernel = { path = "../../kernel" }
+litex = { path = "../../chips/litex" }
+litex_vexriscv = { path = "../../chips/litex_vexriscv" }
```

Comment:
For most boards in Tock that use a chip that depends on some collection of shared libraries (e.g. `nrf52840` depends on the shared libraries in `nrf52`), the more specific crate simply re-exports the portions of the shared crate that might be needed outside the more specific crate. (e.g. `chips/nrf52840` re-exports much of `chips/nrf52`). This allows for boards to depend only on the more specific crate, rather than on both the more specific and less specific crates. I have found that this design simplifies refactoring -- if nrf52840 previously used the generic nrf52 spi library, but is later updated to use an nrf52840-specific version, I do not have to modify every board to reflect the change.

Would it be a good idea to continue that approach here? Or is there some reason that doing so is impossible for the litex chips?

Reply:
I wasn't aware of this approach, but it makes sense. I'll happily update it.

Reply:
Thanks! Re: awareness, thats probably because the approach isn't documented anywhere yet. We probably should try to codify how we support chip families in Tock soon. When it was just nrf52 it didn't make sense, but now we have similar patterns for nrf52, stm32, lowrisc, imxrt, and soon litex.

Reply:
I've updated the board to only depend on the `litex_vexriscv`-crate, which now exposes both the chip module (which is depending on the specific CPU architecture used, in our case VexRiscv) and the rest of the LiteX core drivers and support modules. Feel free to resolve if the current design matches the other chips as requested.